github/localrepo.go

```diff
@@ -118,6 +113,23 @@ func (r *GitHubRepo) MasterBranch() (branch *Branch) {
 	return
 }
 
+func (r *GitHubRepo) RemoteMasterBranch() (b *Branch, ok bool) {
```

Comment:
1. Could this simply return `error` instead of `ok bool`?
2. Why did the original `MasterBranch` needed to change? I don't see anything new in the updated logic. What was the rationale for adding `RemoteMasterBranch`?

Reply:
> 1. Could this simply return error instead of ok bool?

Sure, that would work as well. I guess the question is, what's the more correct signal?

> 2. Why did the original MasterBranch needed to change? I don't see anything new in the updated logic. What was the rationale for adding RemoteMasterBranch?

I wanted to differentiate between:

1. `git.BranchAtRef("refs", "remotes", origin.Name, "HEAD")` returns `refs/heads/master`
2. Above fails and name is set to `refs/heads/master` in `MasterBranch()`.

In the first case, when repo default branch differs, we show a warning. In the second case, we don't.

The new function was added to avoid changing behavior of `MasterBranch`, but I could of course either change the return signature of `MasterBranch` instead and update the code in all relevant locations or remove the fallback `refs/heads/master`. Thoughts?

Reply:
I see, thanks for the explanation. I'm fine with the current approach for now. I would love to see some tests!